src/middlewared/middlewared/plugins/virt/instance_device.py

```diff
@@ -539,16 +539,11 @@ async def set_bootable_disk(self, id, disk):
 
         device_list = await self.device_list(id)
         desired_disk = None
-        max_boot_priority_device = None
 
-        for device_entry in device_list:
-            if (max_boot_priority_device is None and device_entry.get('boot_priority') is not None) or (
-                (device_entry.get('boot_priority') or 0) > ((max_boot_priority_device or {}).get('boot_priority') or 0)
-            ):
-                max_boot_priority_device = device_entry
-
-            if device_entry['name'] == disk:
-                desired_disk = device_entry
+        max_boot_priority_device = get_max_boot_priority_device(device_list)
```

Comment:
I don't understand these changes? `get_max_boot_priority_device` iterates over `device_list` and sets the `max_boot_priority_device`. But right below this, we're iterating the `device_list` again and setting a `desired_disk`?